pkg/envoy/sds/response.go

```diff
@@ -23,59 +24,86 @@ var directionMap = map[envoy.SDSCertType]string{
 }
 
 // NewResponse creates a new Secrets Discovery Response.
-func NewResponse(catalog catalog.MeshCataloger, proxy *envoy.Proxy, request *xds_discovery.DiscoveryRequest, cfg configurator.Configurator, certManager certificate.Manager) (*xds_discovery.DiscoveryResponse, error) {
+func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, request *xds_discovery.DiscoveryRequest, cfg configurator.Configurator, certManager certificate.Manager) (*xds_discovery.DiscoveryResponse, error) {
 	log.Info().Msgf("Composing SDS Discovery Response for proxy: %s", proxy.GetCommonName())
 
-	svcList, err := catalog.GetServicesFromEnvoyCertificate(proxy.GetCommonName())
-	// Github Issue #1575
-	serviceForProxy := svcList[0]
+	svcList, err := meshCatalog.GetServicesFromEnvoyCertificate(proxy.GetCommonName())
 	if err != nil {
-		log.Error().Err(err).Msgf("Error looking up service for Envoy with CN=%q", proxy.GetCommonName())
+		log.Error().Err(err).Msgf("Error getting services associated with proxy %s", proxy.GetCommonName())
 		return nil, err
 	}
 
-	cert, err := certManager.IssueCertificate(serviceForProxy.GetCommonName(), cfg.GetServiceCertValidityPeriod())
+	// OSM currently relies on kubernetes ServiceAccount for service identity
+	svcAccount, err := catalog.GetServiceAccountFromProxyCertificate(proxy.GetCommonName())
 	if err != nil {
-		log.Error().Err(err).Msgf("Error issuing a certificate for proxy service %s", serviceForProxy)
+		log.Error().Err(err).Msgf("Error retrieving SerivceAccount for proxy %s", proxy.GetCommonName())
 		return nil, err
 	}
 
-	// Iterate over the list of tasks and create response structs to be
-	// sent to the proxy that made the discovery request
+	sdsImpl := newSDSImpl(proxy, meshCatalog, certManager, cfg, svcList, svcAccount)
```

Comment:
Not sure what `impl` means here or anywhere below really. I would prefer to use something that reads easily in plain english and helps you understand what's behind this variable.

No need to change this PR - but eventually would be good to add clarity so it is easy to read.

Reply:
I left a comment on what `sdsImpl` refers to in `sds/types.go`.